index.js

```diff
@@ -155,7 +155,24 @@ ShareDbMongo.prototype._connect = function(mongo, options) {
     mongo(finish);
     return;
   }
-  mongodb.connect(mongo, options, finish);
+  // Get mongo options from the explicitly defined field `mongoOptions` or
+  // sanitize the whole options object by removing all ShareDB-specific options
+  // TODO: Deprecate the ability to pass mongo options implicitly
+  var mongoOptions;
+  if (options.mongoOptions) {
+    mongoOptions = options.mongoOptions;
```

Comment:
I like this change of being explicit about which options to pass to mongo in a separate object, but let's not do the second part where we try and make the old options style backward compatible. I think it is reasonable to expect people to update their connection instantiation code as they switch drivers.